core/store/src/trie/iterator.rs

```diff
@@ -342,8 +368,67 @@ mod tests {
         let want: Vec<_> = map
             .range(seek_key.to_vec()..)
             .map(|(k, v)| (k.clone(), v.clone()))
-            .take(5)
             .filter(|(x, _)| x.starts_with(seek_key))
+            .take(5)
```

Comment:
the prior order seemed incorrect?

Reply:
The test isn't competent enough. I tried to change the order of the take and filter and in both cases we generate exactly the same set of changes.

I fear we may need better tests here. The max map size is just 10 elements and in practice I didn't see it go beyond 8.

Reply:
Ok, sorry, ignore this comment, I think the order of the take and filter doesn't matter here as the keys are sorted.

Reply:
modified it to just check all the elements per our discussion.